interpret.js

```diff
@@ -12,10 +12,10 @@ const find = (xs, f) => {
   return found;
 }
 
-const dispatch = pairs =>
+const dispatch = of => pairs =>
```

Comment:
Why should `dispatch` be curried? it kinda didn't makes sense for me, as I can't imagine any scenario where user wants to create dispatcher by providing `of` and `pairs` separately.